aequilibrae/project/project_creation.py

```diff
@@ -0,0 +1,218 @@
+
+        cursor.execute(sql, args)
+
+    conn.commit()
+
+
+def populate_meta_extra_attributes(conn) -> None:
+    fields = []
+    fields.append(['link_types', 'link_type', 'Link type name. E.g. arterial, or connector'])
+    fields.append(['link_types', 'link_type_id', 'Single letter identifying the mode. E.g. a, for arterial'])
+    fields.append(['link_types', 'description',
+                   'Description of the same. E.g. Arterials are streets like AequilibraE Avenue'])
+    fields.append(['link_types', 'lanes', 'Default number of lanes in each direction. E.g. 2'])
+    fields.append(['link_types', 'lane_capacity', 'Default vehicle capacity per lane. E.g.  900'])
+
+    fields.append(['modes', 'mode_name', 'Link type name. E.g. arterial, or connector'])
+    fields.append(['modes', 'mode_id', 'Single letter identifying the mode. E.g. a, for arterial'])
+    fields.append(
+        ['modes', 'description', 'Description of the same. E.g. Arterials are streets like AequilibraE Avenue'])
+
+    extra_keys = ['alpha', 'beta', 'gamma', 'delta', 'epsilon', 'zeta', 'iota', 'sigma', 'phi', 'tau']
+    extra_keys = [[x, 'Available for user convenience'] for x in extra_keys]
+
+    cursor = conn.cursor()
+    for table_name, f, d in fields:
+        sql = f"INSERT INTO '{meta_table}' (name_table, attribute, description) VALUES('{table_name}','{f}', '{d}')"
+        cursor.execute(sql)
+
+    for table_name in ['link_types', 'modes']:
+        for f, d in extra_keys:
+            sql = f"INSERT INTO '{meta_table}' (name_table, attribute, description) VALUES('{table_name}','{f}', '{d}')"
+            cursor.execute(sql)
+    conn.commit()
+
+
+def create_network_tables(conn, parameters) -> None:
+    """Creates empty network tables for future filling"""
+    curr = conn.cursor()
+    # Create the links table
+    fields = parameters["network"]["links"]["fields"]
+
+    sql = """CREATE TABLE 'links' (
+                      ogc_fid INTEGER PRIMARY KEY,
+                      link_id INTEGER UNIQUE,
+                      a_node INTEGER,
+                      b_node INTEGER,
+                      direction INTEGER NOT NULL DEFAULT 0,
+                      distance NUMERIC,
+                      modes TEXT NOT NULL,
+                      link_type TEXT NOT NULL DEFAULT 'default',
+                      {}
+                      FOREIGN KEY (link_type)
+                            REFERENCES link_types (link_type));"""
+
+    flds = fields["one-way"]
+
+    # returns first key in the dictionary
+    def fkey(f):
+        return list(f.keys())[0]
+
+    owlf = ["{} {}".format(fkey(f), f[fkey(f)]["type"]) for f in flds if fkey(f).lower() not in req_link_flds]
+
+    flds = fields["two-way"]
+    twlf = []
+    for f in flds:
+        nm = fkey(f)
+        tp = f[nm]["type"]
+        twlf.extend([f"{nm}_ab {tp}", f"{nm}_ba {tp}"])
+
+    link_fields = owlf + twlf
```

Comment:
I feel that this is very opaque and could be commented / variable named better

Reply:
Improved a bit...